qiskit/transpiler/passes/optimization/hoare_opt.py

```diff
@@ -203,7 +203,10 @@ def _traverse_dag(self, dag):
         """
         import z3
 
-        for node in dag.topological_op_nodes():
+        # It does not look safe to iterate over DAG while removing and replacing
+        # some of its nodes.
```

Comment:
Just an optional suggestion to rephrase this for outside the context of this PR 🙂 
```suggestion
        # Pre-generate all DAG nodes, since we later iterate over them while
        # potentially removing some of them.
```

Reply:
Done in [d819a42](https://github.com/Qiskit/qiskit/pull/13083/commits/d819a424df9297eb2792dce77ef4079f9a9f8c17).